systemd/stratis-fstab-setup-v2

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+if [ -z "$1" ]; then
+	echo Stratis symlink with pool name and filesystem name in systemd format \
+	\(dev-stratis-\$POOL_NAME-\$FILESYSTEM_NAME\) required as an argument. >&2
+	exit 1
+fi
+
+
+if [ "$1" != dev-stratis* ]; then
+	echo Argument provided was in incorrect format. Stratis symlink with pool name \
+	and filesystem name in systemd format \(dev-stratis-\$POOL_NAME-\$FILESYSTEM_NAME\) \
+	required as an argument. >&2
+	exit 1
+fi
+
+PATH="$1"
+
+IFS='-' read -ra SEGMENTS <<< "$PATH"
+
+if [ "${#SEGMENTS[@]}" != 4 ]; then
```

Comment:
This won't work if there are embedded '-' characters in the pool or filesystem name (e.g. p1/fs1-snap). Systemd escapes '-' as `\x2d` in unit names but by the time the argument is passed to the stratis-fstab-setup script the escaping has been expanded to the literal '-' character.